packages/iov-lisk/src/liskconnection.ts

```diff
@@ -206,8 +206,13 @@ export class LiskConnection implements BcpConnection {
     return accounts.length > 0 ? accounts[0] : undefined;
   }
 
-  public getNonce(_: BcpAddressQuery | BcpPubkeyQuery): Promise<Nonce> {
-    return Promise.resolve(generateNonce());
+  public async getNonce(_: BcpAddressQuery | BcpPubkeyQuery): Promise<Nonce> {
+    return generateNonce();
+  }
+
+  public async getNonces(_: BcpAddressQuery | BcpPubkeyQuery, count: number): Promise<ReadonlyArray<Nonce>> {
+    const checkedCount = new Uint53(count).toNumber();
+    return Array.from({ length: checkedCount }).map(_1 => generateNonce());
```

Comment:
Is there an issue with using the same nonce in multiple transactions?
Like should one insert a second offset between each one?
It seems this will generate the same value multiple times

Reply:
Indeed, same transaction content in the same second will lead to the same transaction ID. I actually had this issue when working with the Lisk faucet.

Incrementing the nonce is also not optimal in this case it changes the creation timestamp (which has no real value anyway as you can put anything in there). However, Lisk nodes do not accept timestamps in the future. But it can be [now-3, now-2, now-1, now] for 4 nonces.

Reply:
> But it can be [now-3, now-2, now-1, now] for 4 nonces.

Yes, I prefer something like that to avoid hard to debug errors further down the line

Reply:
This feature makes `getNonces(query, 2)` more advanced than `getNonce(); gerNonce()` because the later gives you two equal nonces

Reply:
done